packages/inter-protocol/src/proposals/replaceElectorate.js

```diff
@@ -0,0 +1,233 @@
+// @ts-check
+import { E } from '@endo/eventual-send';
+import {
+  assertPathSegment,
+  makeStorageNodeChild,
+} from '@agoric/internal/src/lib-chainStorage.js';
+import { reserveThenDeposit } from './utils.js';
+
+const trace = (...args) => console.log('GovReplaceCommiteeAndCharter', ...args);
+
+const { values } = Object;
+
+/** @type {<X, Y>(xs: X[], ys: Y[]) => [X, Y][]} */
+const zip = (xs, ys) => xs.map((x, i) => [x, ys[i]]);
+
+/** @type {(name: string) => string} */
+const sanitizePathSegment = name => {
+  const candidate = name.replace(/[ ,]/g, '_');
+  assertPathSegment(candidate);
+  return candidate;
+};
+
+const handlehighPrioritySendersList = async (
+  { consume: { highPrioritySendersManager: highPrioritySendersManagerP } },
+  { options: { highPrioritySendersConfig } },
+) => {
+  const HIGH_PRIORITY_SENDERS_NAMESPACE = 'economicCommittee';
+  const highPrioritySendersManager = await highPrioritySendersManagerP;
+
+  if (!highPrioritySendersManager) {
+    throw Error('highPrioritySendersManager is not defined');
+  }
+
+  const { addressesToAdd, addressesToRemove } = highPrioritySendersConfig;
+
+  for (const addr of addressesToAdd) {
+    trace(`Adding ${addr} to High Priority Senders list`);
+    await E(highPrioritySendersManager).add(
+      HIGH_PRIORITY_SENDERS_NAMESPACE,
+      addr,
+    );
+  }
```

Comment:
```suggestion
    await Promise.all(addressesToAdd.map(addr => E(highPrioritySendersManager).add(
      HIGH_PRIORITY_SENDERS_NAMESPACE,
      addr,
    )));
```

If the tracing is important, consider:

```js
const traced = (label, x) => {
  trace(label, x);
  return x;
```

and then

```suggestion
    await Promise.all(addressesToAdd.map(addr => E(highPrioritySendersManager).add(
      HIGH_PRIORITY_SENDERS_NAMESPACE,
      traced('High Priority Senders: adding', addr),
    )));
```